dotnet/src/Connectors/Connectors.OpenAI/TextToImage/AzureOpenAITextToImageService.cs

```diff
@@ -3,215 +3,151 @@
         }
 
-        // Defaults if not supplied
-        maxRetryCount ??= 5;
-        apiVersion ??= "2023-06-01-preview";
-
-        this._core = new(httpClient, loggerFactory?.CreateLogger(typeof(AzureOpenAITextToImageService)));
-
-        this._endpoint = !string.IsNullOrEmpty(endpoint) ? endpoint! : httpClient!.BaseAddress!.AbsoluteUri;
-        this._apiKey = apiKey;
-        this._maxRetryCount = maxRetryCount.Value;
-        this._apiVersion = apiVersion;
-        this._core.AddAttribute(AIServiceExtensions.EndpointKey, endpoint);
-        this._core.AddAttribute(AIServiceExtensions.ModelIdKey, modelId);
-        this._core.AddAttribute(AIServiceExtensions.ApiVersionKey, apiVersion);
+        var connectorEndpoint = !string.IsNullOrWhiteSpace(endpoint) ? endpoint! : httpClient?.BaseAddress?.AbsoluteUri;
+        if (connectorEndpoint is null)
+        {
+            throw new ArgumentException($"The {nameof(httpClient)}.{nameof(HttpClient.BaseAddress)} and {nameof(endpoint)} are both null or empty. Please ensure at least one is provided.");
+        }
 
-        this._core.RequestCreated += (_, request) => request.Headers.Add("api-key", this._apiKey);
+        this._client = new(new Uri(connectorEndpoint),
+            new AzureKeyCredential(apiKey),
+            GetClientOptions(httpClient, apiVersion));
     }
 
     /// <inheritdoc/>
-    public IReadOnlyDictionary<string, object?> Attributes => this._core.Attributes;
+    public IReadOnlyDictionary<string, object?> Attributes { get; } = new Dictionary<string, object?>();
 
     /// <inheritdoc/>
-    public async Task<string> GenerateImageAsync(string description, int width, int height, Kernel? kernel = null, CancellationToken cancellationToken = default)
+    public async Task<string> GenerateImageAsync(
+        string description,
+        int width,
+        int height,
+        Kernel? kernel = null,
+        CancellationToken cancellationToken = default)
     {
-        var operationId = await this.StartTextToImageAsync(description, width, height, cancellationToken).ConfigureAwait(false);
-        var result = await this.GetTextToImageResultAsync(operationId, cancellationToken).ConfigureAwait(false);
+        Verify.NotNull(description);
+        const string SizeErr = "Dall-E 2 can generate only square images of size 256x256, 512x512, or 1024x1024";
 
-        if (result.Result is null)
+        if (width != height)
         {
-            throw new KernelException("Azure OpenAI Text To Image null response");
+            throw new ArgumentOutOfRangeException("width,height", $"{width}x{height}", SizeErr);
         }
 
-        if (result.Result.Images.Count == 0)
+        var size = width switch
         {
-            throw new KernelException("Azure OpenAI Text To Image result not found");
+            256 => ImageSize.Size256x256,
+            512 => ImageSize.Size512x512,
+            1024 => ImageSize.Size1024x1024,
```

Comment:
This should be updated to handle `ImageSize.Size1024x1024`, `ImageSize.Size1792x1024`, `ImageSize.Size1024x1792`.